fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/repository/WitnessingRepositoryTest.java

```diff
@@ -67,11 +87,20 @@ public class WitnessingRepositoryTest {
           "",
           new ArrayList<>());
 
+  @Rule(order = 0)
+  public final MockitoRule mockitoRule = MockitoJUnit.rule();
+
   @Before
   public void setUp() {
     Application application = ApplicationProvider.getApplicationContext();
-    appDatabase =
-        AppDatabaseModuleHelper.getAppDatabase(ApplicationProvider.getApplicationContext());
```

Comment:
I don't really understand this line, was the problem that it was not using a mock?

Reply:
The Room database instance created was shared among several thread instances running the tests in parallel, thus creating concurrency issues